### DISCUSSION TITLE
Getting an error when trying to update Wiki.JS

### DISCUSSION THREAD

**Discussion by landshark22** [tteck/Proxmox#2449] [Help Needed]
Getting the following error message when trying to update Wiki.js LXC. Get it when using either "update" or the "bash"

✓ Stopped Wikijs
 ✓ Backed up Data
 ✓ Updated Wikijs
 - Restoring Data  /
[ERROR] in line 78: exit code 0: while executing command npm rebuild sqlite3 &> /dev/null

**Comment by landshark22** [comment]
To add to this. Cannot access my site after trying to update. I've tried to search here to find a solution and can't see to find anything, maybe I'm using the wrong terms?

**Reply by tteck** [reply]
Can't replicate 

```
 _       ___ __   _     _     
| |     / (_) /__(_)   (_)____
| | /| / / / //_/ /   / / ___/
| |/ |/ / / ,< / /   / (__  ) 
|__/|__/_/_/|_/_(_)_/ /____/  
                 /___/      
 
 ✓ Stopped Wikijs
 ✓ Backed up Data
 ✓ Updated Wikijs
 ✓ Restored Data
 ✓ Started Wikijs
 ✓ Updated Successfully
root@wikijs:~# 
```
Try the command `systemctl start wikijs` to bring Wiki.js back online.

**Comment by landshark22** [comment]
Thanks for the response. The Command didn't work. Here's the output of systemctl saying wikijs.service failed:
'''
  UNIT                                                              LOAD   ACTIVE SUB       DESCRIPTION                                            
  -.mount                                                           loaded active mounted   Root Mount
  dev-.lxc-proc.mount                                               loaded active mounted   /dev/.lxc/proc
  dev-.lxc-sys.mount                                                loaded active mounted   /dev/.lxc/sys
  dev-full.mount                                                    loaded active mounted   /dev/full
  dev-mqueue.mount                                                  loaded active mounted   POSIX Message Queue File System
  dev-null.mount                                                    loaded active mounted   /dev/null
  dev-ptmx.mount                                                    loaded active mounted   /dev/ptmx
  dev-random.mount                                                  loaded active mounted   /dev/random
  dev-tty.mount                                                     loaded active mounted   /dev/tty
  dev-tty1.mount                                                    loaded active mounted   /dev/tty1
  dev-tty2.mount                                                    loaded active mounted   /dev/tty2
  dev-urandom.mount                                                 loaded active mounted   /dev/urandom
  dev-zero.mount                                                    loaded active mounted   /dev/zero
  proc-cpuinfo.mount                                                loaded active mounted   /proc/cpuinfo
  proc-diskstats.mount                                              loaded active mounted   /proc/diskstats
  proc-loadavg.mount                                                loaded active mounted   /proc/loadavg
  proc-meminfo.mount                                                loaded active mounted   /proc/meminfo
  proc-slabinfo.mount                                               loaded active mounted   /proc/slabinfo
  proc-stat.mount                                                   loaded active mounted   /proc/stat
  proc-swaps.mount                                                  loaded active mounted   /proc/swaps
  proc-sys-net.mount                                                loaded active mounted   /proc/sys/net
  proc-sysrq\x2dtrigger.mount                                       loaded active mounted   /proc/sysrq-trigger
  proc-uptime.mount                                                 loaded active mounted   /proc/uptime
  run-credentials-systemd\x2dsysctl.service.mount                   loaded active mounted   /run/credentials/systemd-sysctl.service
  run-credentials-systemd\x2dsysusers.service.mount                 loaded active mounted   /run/credentials/systemd-sysusers.service
  run-credentials-systemd\x2dtmpfiles\x2dsetup.service.mount        loaded active mounted   /run/credentials/systemd-tmpfiles-setup.service
  run-credentials-systemd\x2dtmpfiles\x2dsetup\x2ddev.service.mount loaded active mounted   /run/credentials/systemd-tmpfiles-setup-dev.service
  sys-devices-system-cpu.mount                                      loaded active mounted   /sys/devices/system/cpu
  sys-devices-virtual-net.mount                                     loaded active mounted   /sys/devices/virtual/net
  sys-fs-fuse-connections.mount                                     loaded active mounted   FUSE Control File System
* sys-kernel-config.mount                                           loaded failed failed    Kernel Configuration File System                       
  postfix-resolvconf.path                                           loaded active waiting   Watch for resolv.conf updates and restart postfix
  systemd-ask-password-console.path                                 loaded active waiting   Dispatch Password Requests to Console Directory Watch
  systemd-ask-password-wall.path                                    loaded active waiting   Forward Password Requests to Wall Directory Watch      
  init.scope                                                        loaded active running   System and Service Manager                             
  console-getty.service                                             loaded active running   Console Getty
  container-getty@1.service                                         loaded active running   Container Getty on /dev/tty1
  container-getty@2.service                                         loaded active running   Container Getty on /dev/tty2
  cron.service                                                      loaded active running   Regular background program processing daemon
  dbus.service                                                      loaded active running   D-Bus System Message Bus
  ifupdown-pre.service                                              loaded active exited    Helper to synchronize boot up for ifupdown
  ifupdown-wait-online.service                                      loaded active exited    Wait for network to be configured by ifupdown
  networking.service                                                loaded active exited    Raise network interfaces
  nftables.service                                                  loaded active exited    nftables
  postfix.service                                                   loaded active exited    Postfix Mail Transport Agent
  postfix@-.service                                                 loaded active running   Postfix Mail Transport Agent (instance -)
  ssh.service                                                       loaded active running   OpenBSD Secure Shell server
  systemd-journal-flush.service                                     loaded active exited    Flush Journal to Persistent Storage
  systemd-journald.service                                          loaded active running   Journal Service
  systemd-logind.service                                            loaded active running   User Login Management
  systemd-modules-load.service                                      loaded active exited    Load Kernel Modules
  systemd-network-generator.service                                 loaded active exited    Generate network units from Kernel command line
  systemd-networkd.service                                          loaded active running   Network Configuration
  systemd-remount-fs.service                                        loaded active exited    Remount Root and Kernel File Systems
  systemd-sysctl.service                                            loaded active exited    Apply Kernel Variables
  systemd-sysusers.service                                          loaded active exited    Create System Users
  systemd-tmpfiles-setup-dev.service                                loaded active exited    Create Static Device Nodes in /dev
  systemd-tmpfiles-setup.service                                    loaded active exited    Create Volatile Files and Directories
  systemd-update-utmp.service                                       loaded active exited    Record System Boot/Shutdown in UTMP
  systemd-user-sessions.service                                     loaded active exited    Permit User Sessions
* wikijs.service                                                    loaded failed failed    Wiki.js                                                
  -.slice                                                           loaded active active    Root Slice
  system-container\x2dgetty.slice                                   loaded active active    Slice /system/container-getty
  system-modprobe.slice                                             loaded active active    Slice /system/modprobe
  system-postfix.slice                                              loaded active active    Slice /system/postfix
  system.slice                                                      loaded active active    System Slice
  user.slice                                                        loaded active active    User and Session Slice                                 
  dbus.socket                                                       loaded active running   D-Bus System Message Bus Socket
  ssh.socket                                                        loaded active running   OpenBSD Secure Shell server socket
  systemd-initctl.socket                                            loaded active listening initctl Compatibility Named Pipe
  systemd-journald-dev-log.socket                                   loaded active running   Journal Socket (/dev/log)
  systemd-journald.socket                                           loaded active running   Journal Socket
  systemd-networkd.socket                                           loaded active running   Network Service Netlink Socket                         
  basic.target                                                      loaded active active    Basic System
  cryptsetup.target                                                 loaded active active    Local Encrypted Volumes
  getty.target                                                      loaded active active    Login Prompts
  graphical.target                                                  loaded active active    Graphical Interface
  integritysetup.target                                             loaded active active    Local Integrity Protected Volumes
  local-fs-pre.target                                               loaded active active    Preparation for Local File Systems
  local-fs.target                                                   loaded active active    Local File Systems
  multi-user.target                                                 loaded active active    Multi-User System
  network-online.target                                             loaded active active    Network is Online
  network-pre.target                                                loaded active active    Preparation for Network
  network.target                                                    loaded active active    Network
  paths.target                                                      loaded active active    Path Units
  remote-cryptsetup.target                                          loaded active active    Remote Encrypted Volumes
  remote-fs.target                                                  loaded active active    Remote File Systems
  remote-veritysetup.target                                         loaded active active    Remote Verity Protected Volumes
  slices.target                                                     loaded active active    Slice Units
  sockets.target                                                    loaded active active    Socket Units
  swap.target                                                       loaded active active    Swaps
  sysinit.target                                                    loaded active active    System Initialization
  time-set.target                                                   loaded active active    System Time Set
  timers.target                                                     loaded active active    Timer Units
  veritysetup.target                                                loaded active active    Local Verity Protected Volumes                         
  apt-daily-upgrade.timer                                           loaded active waiting   Daily apt upgrade and clean activities
  apt-daily.timer                                                   loaded active waiting   Daily apt download activities
  dpkg-db-backup.timer                                              loaded active waiting   Daily dpkg database backup timer
  e2scrub_all.timer                                                 loaded active waiting   Periodic ext4 Online Metadata Check for All Filesystems
  logrotate.timer                                                   loaded active waiting   Daily rotation of log files
  man-db.timer                                                      loaded active waiting   Daily man-db regeneration
  systemd-tmpfiles-clean.timer                                      loaded active waiting   Daily Cleanup of Temporary Directories '''

**Reply by tteck** [reply]
🤷🏻‍♂️  Restore a backup

**Reply by odolphie** [reply]
I've also now restored from a backup as the same error as OP.  Can't figure out whats occuring as the files seem to get copied into the backup folder, copied back fine but the `npm rebuild sqlite3 &> /dev/null` errors out every time.

**Reply by landshark22** [reply]
@odolphie mind if I asl where the backup folder is located and how to restore?

**Reply by landshark22** [reply]
Actually, I think I may have found it /var/backups ? 

If so not sure how to restore.

**Reply by odolphie** [reply]
Ah i see the confusion, talking about two types of backup.  Restoring from the LXC backup, and the backup thats part of the update script.  To get back up and running I just restored the whole LXC from my nightly backups. https://pve.proxmox.com/wiki/Backup_and_Restore

The rest of my comment was related to the script itself, it takes a backup of the database, does the upgrade and copies it back.  The copying part seems to be going fine, the issue lies with the sqlite3 rebuild.
